task/google/resources/common.go

```diff
@@ -0,0 +1,39 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"google.golang.org/api/compute/v1"
+	"google.golang.org/api/googleapi"
+)
+
+func waitForOperation(ctx context.Context, timeout time.Duration, minimum time.Duration, maximum time.Duration, function func(...googleapi.CallOption) (*compute.Operation, error), arguments ...googleapi.CallOption) (*compute.Operation, error) {
+	for delay, deadline := minimum, time.Now().Add(timeout); time.Now().Before(deadline); delay *= 2 {
+		switch operation, err := function(arguments...); {
+		case err == nil && operation.Status == "DONE":
+			return operation, nil
+
+		case err == nil && operation.Error != nil:
+			return nil, fmt.Errorf("operation error: %#v", *operation.Error.Errors[0])
+
+		case err != nil && !strings.HasSuffix(err.Error(), "resourceNotReady"):
+			return nil, err
+
+		default:
+			log.Printf("waiting for operation: (%#v, %#v)", operation, err)
+		}
+
+		if delay < maximum {
```

Comment:
Because. Period. 😉 

```go
cannot use ··· (type time.Duration) as type float64 in argument to math.Min
```

Reply:
I'd wish!

Reply:
golang... they said its cool

Reply:
if its a number you could build the duration later on maybe? Anyway its not a big deal, only the extra super verbosity of golang

Reply:
I could, but that's the most “idiomatic” approach I know of. Go may be cool, but not because of this, or, at least, not for people like us who prefer a terser syntax.